app/models/order_detail.rb

```diff
@@ -927,6 +928,20 @@ def resolve_dispute?
     ActiveModel::Type::Boolean.new.cast(resolve_dispute)
   end
 
+  def update_completed_fulfilled_at
```

Comment:
I think you've got the `@editing_user` in the param updater.  If you pass that in, could you do the logging hare as part of this method?  That might simplify things a bit.
```suggestion
  def update_completed_fulfilled_at(@editing_user)
```